### ISSUE TITLE
Flexible frame configurations (e.g. intersection/union handling)

### ISSUE THREAD

**Issue by herjy** [pmelchior/scarlet#120]
At the moment, multi resolution reconstructs the scene inside the footprint of the high resolution image, while accounting for all the low resolution pixels. This works best if the high resolution frame is fully contained in the low resolution one.
We should leave it as an option for users to reconstruct scenes in the union or overlap of all frames across resolutions.

**Comment by pmelchior** [comment]
It's a more general issue: we assume that the frame of Blend is contained in the data, even if all frames have the same resolution.

We should indeed have the ability to chose the data and blend frames flexibly, which would account for union/intersections. Renamed the issue to clarify that.

**Comment by pmelchior** [comment]
Beyond this issue, we should think about what really constitutes the blend frame. In essence, it's a piece of the sky model. How to segment it is entirely non-trivial, but necessary to run at survey scale.

**Comment by herjy** [comment]
Being able to make models that live in the union of two observations rose a chicken and egg problem of the sort. the union requires that we know the common `frame` that encapsulates 2 `observations`. But the `frame` is defined outside the `observation` class and therefore cannot know the size of the picture that encapsulates both observations. 
What I propose is that we create a class `MultiObservations` that would take as arguments a set of observations and a reference psf and that would return the arguments to set up the `frame`. Then this `frame` could be matched to the `observations` by calling `MultiObservation.match(frame)`. Alternativelly, `Frame_MultiObservations` could inherit from frame, generate the frame and operate the matchings at the same time. This way, the whole 5th block in notebook `tutorials/multiresolution.ipynb` would be replaced by:
```
obs_hst = scarlet.Observation(data_hst, wcs=wcs_hst, psfs=psf_hst, channels=channels_hst)
obs_hsc = scarlet.LowResObservation(data_hsc,  wcs=wcs_hsc, psfs=psf_hsc, channels=channels_hsc)
frame = Frame_MultiObservations([obs_hsc, obs_hst], psf_hst, wcs_hst)
```
This is a small change in the user interface, but it only affects the multi-resolution usecase, and it makes things, I think, clearer from the user's point of view.

**Comment by pmelchior** [comment]
I'm thinking of something similar, but without the need for a special class. Instead of line 3 in your example, I previously suggested a class method:
```
frame = Frame.from_observations(observations, perimeter='intersection')
```
where `perimeter in ['intersection', 'union']`.

This method *may* also decide what wcs and PSF is should have, e.g. by simply picking the finest-resolution wcs and the sharpest PSF, or we may set them afterwards, by adding
```
frame.psfs = psfs
```
I prefer this option because there is no other functionality beside the initial choice of footprints that cannot be done by the default `Frame` implementation. That's ideal for a class method. It would also align with another (future) class method
```
frame = Frame.from_survey("HSC", tile_id=104928)
```
for scanning a full survey one frame at a time.

**Comment by herjy** [comment]
I was aiming at not making frame too complex as it is used in other use modes of scarlet than multi-resolution. Also, I had this chicken and egg conceptual problem that frame would have a method that uses observations as arguments and observations has a method that uses frame as argument. It is not a practical problem, but conceptually, I find that twisted.

**Comment by herjy** [comment]
Addressed as part of PR #130
